account_analytic_tag_distribution/models/account_move.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2023 Tecnativa - Víctor Martínez
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _set_analytic_distribution_from_analytic_tags(self):
+        for item in self.invoice_line_ids.filtered(lambda x: x.analytic_tag_ids):
+            tags = item.analytic_tag_ids.filtered(
+                lambda x: x.active_analytic_distribution
+            )
+            if tags:
+                item_distribution = item.analytic_distribution or {}
+                for distribution in tags.mapped("analytic_distribution"):
+                    for d_key in list(distribution.keys()):
+                        if d_key not in item_distribution:
+                            item_distribution[d_key] = distribution[d_key]
+                        else:
+                            item_distribution[d_key] += distribution[d_key]
+                item.analytic_distribution = item_distribution
+
+    def action_post(self):
+        """Before post, set the analytical distribution according to analytic tags."""
```

Comment:
Why before post?

Reply:
In that method call `_create_analytic_lines()` method to create the analytic items, I think it is better to use `_create_analytic_lines()` directly, I will change it.